frontend/optmath.lua

```diff
@@ -102,4 +104,25 @@ function Math.tmax(tab, func)
     return tmin_max(tab, func, "max")
 end
 
+--[[--
+Restricts a value within an interval.
+
+@tparam value value
+@tparma min minimum
+@tparam max maximum
+@treturn a value clipped to the interval [min, max]
+]]
+function Math.minmax(value, min, max)
```

Comment:
Isn't `clamp` a standard name for that?

Reply:
In signal-processing if a value gets bigger than it should it gets clipped. But I am no English doctor.

Reply:
+1 for clamp ;).

Reply:
> Isn't clamp a standard name for that?

minmax() is a fairly standard name too, but you might be right that clamp() is more standard for this specific behavior.

Reply:
So, with clamp() being less explicite that minmax() for its argument order, should it works the same whether `clamp(value, 9, 3)` or `clamp(value, 3, 9)` ? or should it still error in the first case ?

Reply:
Actually that would be clamp(min, val, max)

Reply:
Oh, C++ does (val, min, max), but https://developer.mozilla.org/en-US/docs/Web/CSS/clamp()

Reply:
To avoid mix-ups (as we are here programmers and css tweakers), I want to suggest we use it like 'Math.clamp(val, {min, max})'.

Reply:
Another way is `Math.mid(x, y, z)` returning middle value independently of the arguments order.

Reply:
> To avoid mix-ups (as we are here programmers and css tweakers), I want to suggest we use it like `Math.clamp(val, {min, max})`.

That would create a Lua table for each call, which would be uneedlessly more expensive, if we were to use it in loops.

I think we're just fine with `clamp(val, min, max)` which is how you naturally wrote it and how I would intuitively expect it to work :)

Reply:
> Another way is Math.mid(x, y, z) returning middle value independently of the arguments order.

Clever, it seems to do what clamp() does - but it needed me to think and try in my head what the result would be with various cases - so no :) expliciteness/clarity > cleverness :)

Reply:
> To avoid mix-ups (as we are here programmers and css tweakers), I want to suggest we use it like 'Math.clamp(val, {min, max})'.

(val, min, max) is fine